corehq/apps/export/views.py

```diff
@@ -485,9 +446,112 @@ def _check_export_size(self, export_instances, filters):
                 }
             )
 
-    def _get_mobile_user_and_group_slugs(self, filter_slug):
-        matches = self.mobile_user_and_group_slugs_regex.findall(filter_slug)
-        return [n[1] for n in matches]
+        # Generate filename
+        if len(export_instances) > 1:
+            filename = "{}_custom_bulk_export_{}".format(domain, date.today().isoformat())
+        else:
+            filename = "{} {}".format(export_instances[0].name, date.today().isoformat())
+
+        download = get_export_download(
+            export_instances=export_instances,
+            filters=export_filters,
+            filename=filename,
+        )
+    except ExportAsyncException as e:
+        return json_response({
+            'error': str(e),
+        })
+    except XlsLengthException:
+        return json_response({
+            'error': _('This file has more than 256 columns, which is not supported by xls. '
+                       'Please change the output type to csv or xlsx to export this file.')
+        })
+    except Exception:
+        return json_response({
+            'error': _("There was an error."),
+        })
```

Comment:
The try/except blocks in here could stand to be cleaned up - they're nested, for one, and have a very broad scope.  In particular, I think by catching `Exception`, we won't get sentry error if there's a bug here.

I'd like to see these exceptions wrapping only the scope that's expected to fail, and if there's an unexpected exception, that we let this 500.  The JS code can check the status code of the response.  We do also have a `@json_error` decorator which returns a json response and still logs the error.

Reply:
Removed `except Exception` in fc9b161a803  The js code already handles unexpected exceptions.

Reply:
Broke the big block into smaller blocks, so only the exception-throwing code is in a `try`, in 43ac2503bcc